src/lightning/fabric/CHANGELOG.md

```diff
@@ -41,12 +41,15 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
   * Handles initialization for FSDP models before wrapping and the Zero stage 3 initialization for DeepSpeed before sharding
 
 
-- Added supports for empty weight initialization with `Fabric.init_module(empty_init=True)` for efficient sharding and checkpoint loading ([#17627](https://github.com/Lightning-AI/lightning/pull/17627))
```

Comment:
Why was this deleted? Please keep it. Of course not all strategies support it, but it is useful for stage 3 training for sure.

Reply:
I thought it was wrong: EmptyInit is not used with DS zero stage 3. It's managed by DeepSpeed internally. With this I mean that the behaviour was is unchanged after the addition of `empty_init`. In the code we have

`super().module_init_context(empty_init=empty_init) if not self.zero_stage_3 else nullcontext()`

Reply:
**our** empty init is not used with stage 3 because it is not needed when the context manager for stage 3 uses the fake tensors anyway. That's why we are setting empty_init=None as default for DeepSpeed and validate the use depending on the configured stage. The empty_init argument in the `init_module` method is not to be directly associated with our _EmptyInit implementation. 

The whole point of all this work was to do this, and now deleting it from the changelog is strange IMO.

Reply:
What I'm saying is that changing `empty_init` does not change whether sharding is more or less efficient today:
- FSDP doesn't support meta device instantiation at the moment
- DeepSpeed Zero stage 3 ignores the `empty_init` argument. It uses fake tensors anyways

In the future, this might not be the case anymore and then it will help with efficient sharding. But today that snippet in the CHANGELOG is misleading